### PR TITLE
pipgrip: add gcc dependency

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See https://github.com/ddelange/pipgrip/issues/36 for more info.

`pipgrip` sometimes needs `gcc` to work properly. This PR adds `gcc` as a dependency as sets up the needed `CC` and `CXX` shell variables.

### CODE REVIEW

File: Formula/pipgrip.rb [hunk 1]
```diff
@@ -47,10 +49,17 @@ class Pipgrip < Formula
   end
 
   def install
-    virtualenv_install_with_resources
+    venv = virtualenv_create(libexec, Formula["python@3.8"].opt_bin/"python3")
+    venv.pip_install resources
+    venv.pip_install buildpath
+
+    gcc_path = Formula["gcc"].bin
```
Comment (dawidd6) [inline Formula/pipgrip.rb:56]:
```suggestion
    gcc_path = Formula["gcc"].opt_bin
```

So it won't break on gcc updates.

File: Formula/pipgrip.rb [hunk 2]
```diff
@@ -47,10 +49,17 @@ class Pipgrip < Formula
...
+    (bin/"pipgrip").write_env_script(libexec/"bin/pipgrip", { CC: gcc_path/"gcc-10", CXX: gcc_path/"g++-10" })
```
Comment (dawidd6) [inline Formula/pipgrip.rb:57]: Plain gcc does not work?

Comment (Rylan12) [reply Formula/pipgrip.rb:57]: Not sure exactly what you mean here

Comment (dawidd6) [reply Formula/pipgrip.rb:57]: The binary name. It's gcc-10 now, could this be just gcc?

Comment (Rylan12) [reply Formula/pipgrip.rb:57]:
```console
$ ls /usr/local/opt/gcc/bin
c++-10                              gfortran-10
cpp-10                              lto-dump-10
g++-10                              x86_64-apple-darwin19-c++-10
gcc-10                              x86_64-apple-darwin19-g++-10
gcc-ar-10                           x86_64-apple-darwin19-gcc-10
gcc-nm-10                           x86_64-apple-darwin19-gcc-10.1.0
gcc-ranlib-10                       x86_64-apple-darwin19-gcc-ar-10
gcov-10                             x86_64-apple-darwin19-gcc-nm-10
gcov-dump-10                        x86_64-apple-darwin19-gcc-ranlib-10
gcov-tool-10                        x86_64-apple-darwin19-gfortran-10
gfortran
```

I don't see a non-versioned option here

Comment (Rylan12) [reply Formula/pipgrip.rb:57]: I could maybe do something like:

```ruby
gcc_version = Formula["gcc"].version.to_s.split(".").first

(bin/"pipgrip").write_env_script(libexec/"bin/pipgrip", { CC: gcc_path/"gcc-#{gcc_version}", CXX: gcc_path/"g++-#{gcc_version}" })
```

Comment (dawidd6) [reply Formula/pipgrip.rb:57]: Could be like that

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**